cmd/artifact-server/main.go

```diff
@@ -0,0 +1,145 @@
+	"golang.org/x/exp/slices"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"gopkg.in/yaml.v3"
+
+	gnostic_metrics_v1 "github.com/google/gnostic/metrics"
+	openapi_v2 "github.com/google/gnostic/openapiv2"
+	openapi_v3 "github.com/google/gnostic/openapiv3"
+)
+
+var messageTypes map[string]proto.Message = map[string]proto.Message{
+	"gnostic.metrics.Complexity":                                   &gnostic_metrics_v1.Complexity{},
+	"gnostic.metrics.VersionHistory":                               &gnostic_metrics_v1.VersionHistory{},
+	"gnostic.metrics.Vocabulary":                                   &gnostic_metrics_v1.Vocabulary{},
+	"gnostic.openapiv2.Document":                                   &openapi_v2.Document{},
+	"gnostic.openapiv3.Document":                                   &openapi_v3.Document{},
+	"google.cloud.apigeeregistry.applications.v1alpha1.Index":      &rpc.Index{},
+	"google.cloud.apigeeregistry.applications.v1alpha1.Lint":       &rpc.Lint{},
+	"google.cloud.apigeeregistry.applications.v1alpha1.LintStats":  &rpc.LintStats{},
+	"google.cloud.apigeeregistry.applications.v1alpha1.References": &rpc.References{},
+	"google.cloud.apigeeregistry.v1.apihub.DisplaySettings":        &rpc.DisplaySettings{},
+	"google.cloud.apigeeregistry.v1.apihub.Lifecycle":              &rpc.Lifecycle{},
+	"google.cloud.apigeeregistry.v1.apihub.ReferenceList":          &rpc.ReferenceList{},
+	"google.cloud.apigeeregistry.v1.apihub.TaxonomyList":           &rpc.TaxonomyList{},
+	"google.cloud.apigeeregistry.v1.controller.Manifest":           &rpc.Manifest{},
+	"google.cloud.apigeeregistry.v1.controller.Receipt":            &rpc.Receipt{},
+	"google.cloud.apigeeregistry.v1.scoring.Score":                 &rpc.Score{},
+	"google.cloud.apigeeregistry.v1.scoring.ScoreCard":             &rpc.ScoreCard{},
+	"google.cloud.apigeeregistry.v1.scoring.ScoreCardDefinition":   &rpc.ScoreCardDefinition{},
+	"google.cloud.apigeeregistry.v1.scoring.ScoreDefinition":       &rpc.ScoreDefinition{},
+	"google.cloud.apigeeregistry.v1.style.ConformanceReport":       &rpc.ConformanceReport{},
+	"google.cloud.apigeeregistry.v1.style.StyleGuide":              &rpc.StyleGuide{},
+	"google.protobuf.FileDescriptorSet":                            &descriptorpb.FileDescriptorSet{},
+}
+
+func handleRequest(w http.ResponseWriter, req *http.Request) {
+	ctx := context.Background()
+	client, err := connection.NewRegistryClient(ctx)
+	if err != nil {
+		writeError(w, err, http.StatusInternalServerError)
+		return
+	}
+	name := strings.TrimPrefix(req.URL.Path, "/")
+	contents, err := client.GetArtifactContents(ctx, &rpc.GetArtifactContentsRequest{Name: name})
+	if err != nil {
+		writeError(w, err, http.StatusNotFound)
+		return
+	}
+	if contents.GetContentType() == "text/plain" {
+		fmt.Fprintf(w, "%s\n", string(contents.GetData()))
```

Comment:
Does the explicit call to `string()` change anything here? I would expect the `fmt` package to do that conversion for us since you're using the `%s` formatter.

Reply:
You're right - I confirmed that the `string()` call can be removed. It feels like magic to me (requiring deepish Go knowledge) but I made the change on your suggestion. Thanks!